packages/records/lib/models/records.ts

```diff
@@ -375,21 +391,36 @@ export async function upsert({
                     .from('upsert')
                     .leftJoin('existing', 'upsert.external_id', 'existing.external_id')
                     .orderBy([
-                        { column: 'updated_at', order: 'asc' },
-                        { column: 'id', order: 'asc' }
+                        { column: 'upsert.updated_at', order: 'asc' },
+                        { column: 'upsert.id', order: 'asc' }
                     ]);
-                const updatedRes = await withRetry(query);
 
-                const inserted = updatedRes.filter((r) => r.status === 'inserted').map((r) => r.external_id);
-                const undeleted = updatedRes.filter((r) => r.status === 'undeleted').map((r) => r.external_id);
-                const deleted = updatedRes.filter((r) => r.status === 'deleted').map((r) => r.external_id);
-                const updated = updatedRes.filter((r) => r.status === 'changed').map((r) => r.external_id);
+                const res = await withRetry(query);
+
+                // Billing:
+                // A record is billed only once per month. ie:
+                // - If a record is inserted, it is billed
+                // - If a record is updated, it is billed if it has not been billed yet during the current month
+                // - If a record is undeleted, it is billed if it has not been billed yet during the current month
+                // - If a record is deleted, it is billed if it has not been billed yet during the current month
```

Comment:
cc @bastienbeurier to make sure I correctly understood the requirements

Reply:
Thanks for pinging me on the details of this. I realize there are multiple options here.

 I would only bill for:

> inserted records are billed
> updated records are billed if they were not inserted/updated in the same calendar month

I would not bill for:

> deleted records are billed if they were not inserted/updated in the same calendar month

I know we have costs when records are deleted. But I think it would be misleading if a customer deleted a connection and was billed for all records for that connection. There would basically be some kind of exit tax for Nango, which I don't think is a good pattern.

> undeleted records are billed if they were not inserted/updated in the same calendar month

This looks like unmodified records to me, so I would not bill for this.

Reply:
Still I would focus on inserting/edition!